src/App.tsx

```diff
@@ -15,33 +15,140 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+export const App: React.FC = () => {
+  const goods = [...goodsFromServer];
+
+  const [isVisible, setVisio] = useState(false);
+  const [isReversed, setReversed] = useState(false);
+  const [isSortByAlphabet, sortAlphabetically] = useState(false);
+  const [isSortByLength, sortByLength] = useState(false);
+  const [isReset, setResetedGoods] = useState(false);
+  const [goodsLength, setGoodsLength] = useState('1');
+  const visibleGoods = goods.filter(good => good.length >= Number(goodsLength));
+
+  switch (true) {
+    case isSortByLength:
+      visibleGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+
+    case isSortByAlphabet:
+      visibleGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+
+    default:
+      break;
+  }
```

Comment:
this solution is weak
I would suggest one more option you could have used: just create handlers and pass them to buttons, and when those button are clicked – perform the action (i.e. sort, reset, show – and so on)